### PR TITLE
Process data models

### PR DESCRIPTION
**Pull request by HenaryKhuraijam**
### Create models
- Create model classes for all entities as shown in the ERD diagram.
- Set up associations between models.
- Remember that author_id column in posts table should be the foreign_key for the users table.
### Use models to insert data
- Familiarize yourself with 'rails c' command.
- Open rails console for your project.
- Create at least one user by running the following code:
  - first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
  - second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
- Create at least 4 posts written by one of the users you created by running the following code:
   - first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
- Create at least 6 posts comments for one of the posts you created by running the following code:
   - Comment.create(post: first_post, user: second_user, text: 'Hi Tom!' )
- Use other CRUD methods to find, update and delete entities.
### Create custom methods
- Your models should include some additional methods.

- Users
   - A method that returns the 3 most recent posts for a given user.
- Posts
   - A method that updates the posts counter for a user.
   - A method which returns the 5 most recent comments for a given post.
- Comments
   - A method that updates the comments counter for a post.
- Likes
   - A method that updates the likes counter for a post.
   - Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_save :update_likes_counter
+
+  def update_likes_counter
+    post.increment!(:comments_counter)
```
Comment (divinecharlotte) [inline app/models/like.rb]: - [ ] kindly consider making sure this method updates  the likes counter 😀  for a post as we are in `likes` counter not `comments_counter ` doing so will make this model meet the project requirements

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @HenaryKhuraijam     , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- Clean code👍 
- Implemented some project requirements   👍 

 




 

## Required Changes ♻️

- [ ] Kindly consider adding a working link to your license under the license section to make your readme more professional and interactive

_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (HenaryKhuraijam) [reply app/models/like.rb]: Done :+1:

**Review by TadesseDev — APPROVED** [review]
### Hi @HenaryKhuraijam 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**